### PR TITLE
Preserve the selected view on scenario result updates

### LINKED ISSUE #1675: Feature request / DAML Studio: Don't revert back to table view on changes
If you have chosen the "transaction view" for showing the result of your scenario, and you make a change that automatically reruns the scenario, the view is reverted back to "table view". 
I don't think the user would expect that. I find it more convenient to stick to the chosen view.

### PR DESCRIPTION
**Pull request by cocreature**
Previously, we always reverted back to the table view when the
scenario results changed. This PR changes this to preserve the
selected view.

I’ve also tested that this does not break backwards compatibility: If
you use the newer extension with an older SDK, you will get the
previous behavior of reverting to the table view and you can still
switch by clicking on the button.

Fixes #1675

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-foundations/daml-tools/daml-extension/src/extension.ts [hunk 1]
```diff
@@ -266,8 +266,12 @@ namespace DamlWorkspaceValidationsNotification {
 }
 
 class VirtualResourceManager {
+    // Mapping from URIs to the web view panel
     private _panels: Map<string, vscode.WebviewPanel> = new Map<string, vscode.WebviewPanel>();
+    // Mapping from URIs to the HTML content of the webview
     private _panelContents: Map<string, string> = new Map<string, string>();
```
Comment (neil-da) [inline daml-foundations/daml-tools/daml-extension/src/extension.ts]: Is there no `Uri` type better than `string`?

File: daml-foundations/daml-tools/daml-extension/src/extension.ts [hunk 2]
```diff
@@ -266,8 +266,12 @@ namespace DamlWorkspaceValidationsNotification {
...
+    // Mapping from URIs to selected view
+    private _panelStates: Map<string, string> = new Map<string, string>();
```
Comment (neil-da) [inline daml-foundations/daml-tools/daml-extension/src/extension.ts]: Selected view as a `string` is also a bit sad.

File: daml-foundations/daml-tools/daml-extension/src/extension.ts (hunk 1 above)
Comment (cocreature) [reply daml-foundations/daml-tools/daml-extension/src/extension.ts]: I actually tried to use `Map<Uri, …>` initially and spend over an hour trying to figure out why it didn’t work until I figured out that this uses reference equality with no way to override it as far as I can tell.

Comment (cocreature) [reply daml-foundations/daml-tools/daml-extension/src/extension.ts]: I’ll add a comment when I touch this again.

File: daml-foundations/daml-tools/daml-extension/src/extension.ts (hunk 2 above)
Comment (cocreature) [reply daml-foundations/daml-tools/daml-extension/src/extension.ts]: Given that we treat it as an opaque value, i.e., we just take what the client gave us and send it back I don’t think it makes sense to actually parse this value in particular given the backwards compat constraints on the IDE. In Haskell i would have used a newtype but searching for newtypes in typescript didn’t bring up anything that looked reasonably nice.

Comment (neil-da) [reply daml-foundations/daml-tools/daml-extension/src/extension.ts]: I think type aliases for uri and html would go a long way to making the comments part of the code, even if they aren't statically checked

File: daml-foundations/daml-tools/daml-extension/src/extension.ts (hunk 1 above)
Comment (neil-da) [reply daml-foundations/daml-tools/daml-extension/src/extension.ts]: I forgot that some languages are completely broken and not fit for purpose...

**Pull request merged**